src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -347,4 +352,19 @@ export class TerminalService implements ITerminalService {
 		}
 		return parts.join('_') + '.UTF-8';
 	}
+}
+
+interface ITerminalProcessConfiguration {
+	_name: string;
+	_shellPath: string;
+}
+
+class TerminalProcessConfiguration implements ITerminalProcessConfiguration {
```

Comment:
We don't really need to implement the interface, an interface is all we need to enforce the types. TypeScript allows regular objects `{}` to use interfaces and keeps the type safe.